DFC.ServiceTaxonomy.GraphSync/GraphSyncers/Parts/BagPartGraphSyncer.cs

```diff
@@ -81,10 +82,31 @@ public BagPartGraphSyncer(IContentDefinitionManager contentDefinitionManager, IS
             return delayedCommands;
         }
 
-        public Task<bool> VerifySyncComponent(ContentItem contentItem, INode node,
+        public async Task<bool> VerifySyncComponent(
+            ContentItem contentItem,
             ContentTypePartDefinition contentTypePartDefinition,
-            IEnumerable<IRelationship> relationships, IEnumerable<INode> destNodes) =>
-            Task.FromResult(true);
+            INode sourceNode,
+            IEnumerable<IRelationship> relationships,
+            IEnumerable<INode> destNodes)
+        {
+            var graphSyncValidator = _serviceProvider.GetRequiredService<IGraphSyncValidator>();
+
+            var contentItems = contentItem.Content[contentTypePartDefinition.Name]["ContentItems"].ToObject<IEnumerable<ContentItem>>();
+
+            foreach (var bagPartContentItem in contentItems)
+            {
+                if (await graphSyncValidator.CheckIfContentItemSynced(bagPartContentItem))
```

Comment:
Apologies if you're still working on this 😄 , but the foreach will never get part the first iteration.